core/domain/user_jobs_one_off.py

```diff
@@ -468,6 +468,148 @@ def reduce(key, values):
         yield (key, len(values))
 
 
+class FixUserSettingsCreatedOnOneOffJob(jobs.BaseMapReduceOneOffJobManager):
+    """Job that fixes the invalid values of created_on attribute in the
+    UserSettingsModel.
+    """
+
+    @classmethod
```

Comment:
I have added it now.

Though I also want to understand in general when we should override the `enqueue` method, i.e when should I increase the `shard_count`. I only understand that it's used to process things in parallel. The test classes using it say that "we can raise the number of shards for this job, since it goes only over one type of entity class", but I found other test classes also going through one entity class type but not using it. Couldn't deduce anything solid from what I observed. Just curious to know. Thanks :)

Reply:
Yep, we implemented this "quite" recently (around 9 months) and only add it to jobs that were created from that time and where we remembered about this.

Reply:
Ok, thanks for the info!